sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -88,23 +100,62 @@ package object debug {
     }
   }
 
+  /**
+   * Get WholeStageCodegenExec subtrees and the codegen in a query plan into one String
+   *
+   * @param query the streaming query for codegen
+   * @return single String containing all WholeStageCodegen subtrees and corresponding codegen
+   */
+  def codegenString(query: StreamingQuery): String = {
+    val msg = query match {
+      case w: StreamExecution if w.lastExecution != null =>
```

Comment:
I think this can be rewritten as if `case w: StreamExecution => if (w.lastExecution != null) ... else ...`

Reply:
Addressed.